crypto/x509/x_all.c

```diff
@@ -138,12 +170,51 @@ X509 *X509_load_http(const char *url, BIO *bio, BIO *rbio, int timeout)
                                    ASN1_ITEM_rptr(X509));
 }
 
+/*
+ * Drop empty SKID or AKID X509_REQ extensions that might result from:
+ * "...KeyIdentifier = none"
+ */
+
+static void drop_empty_req_keyid_exts(X509_REQ *x)
+{
+    STACK_OF(X509_EXTENSION) *exts;
+    size_t i;
+
+    if ((exts = ossl_x509_req_get1_extensions_by_nid(x, NID_ext_req)) == NULL)
+        return;
+
+    for (i = 0; i < OSSL_NELEM(keyid_nids); ++i) {
+        int idx = X509v3_get_ext_by_NID(exts, keyid_nids[i], -1);
+
+        if (idx >= 0) {
+            X509_EXTENSION *ext = X509v3_get_ext(exts, idx);
+            ASN1_OCTET_STRING *der = X509_EXTENSION_get_data(ext);
+
+            /* Suppress AKID "none" */
+            if (ASN1_STRING_length(der) <= 2) {
+                X509v3_delete_ext(exts, idx);
+                X509_EXTENSION_free(ext);
+            }
+        }
+    }
+
+    /* Drop resulting attribute if empty */
+    if (exts != NULL && sk_X509_EXTENSION_num(exts) == 0) {
```

Comment:
```suggestion
    if (sk_X509_EXTENSION_num(exts) == 0) {
```
Here `exts` cannot be NULL, and if it was, `sk_X509_EXTENSION_num(exts)` would return -1.